### ISSUE TITLE
Document the responsibilities of the Controller

### ISSUE THREAD

**Issue by nathanielc** [influxdata/platform#453]

**Comment by nathanielc** [comment]
The controller controls resources as well as the life-cycle of a query. Specifically the controller is what enables queries to be canceled out of band as well as provides visibility into the running queries and their states. In the SQL world the controller is responsible for the `KILL QUERY` and `SHOW QUERIES` queries.

Additionally since the controller controls which resources can be consumed its is also responsible for reporting on those resources. 

As for whether the controller or the executor should manage the resources its the controller. Currently the controller does create the allocator and blocks queries from running until there are enough available resources. The executor just gets to consume the available resources.

As for more complicated ways of managing resources we can investigate those as they arise. 

In summary the controller controls:

* Query Resources - concurrency and memory usage
* Query life-cycle - decides when queries can move to the next step, and can kill/cancel running queries

Part of controlling these things the controller is responsible for providing visibility into the things its controlling.

**Comment by jsternberg** [comment]
That sounds good. One additional question for now. Calculations involve pushing down resources to the storage engine. How do the controller resource limits interact with storage?

**Comment by nathanielc** [comment]
> How do the controller resource limits interact with storage?

Right now they do not. But I expect it will work in a similar fashion. The arrow work requires that we have an allocator. So once that is in place it will be possible to limit the RAM usage of the storage requests. How the controller defines what queryd can use versus storaged is still open for design. We should loop @e-dard into that conversation when we have it.
